eZ/Publish/Core/FieldType/Keyword/KeywordStorage/Gateway/LegacyStorage.php

```diff
@@ -378,4 +387,50 @@ protected function deleteOrphanedKeywords()
 
         $deleteQuery->prepare()->execute();
     }
+
+    /**
+     * @param mixed $fieldId
+     * @param int $versionNo
+     * @return bool
+     */
+    private function publishedVersionForFieldExists($fieldId, $versionNo)
+    {
+        $selectQuery = $this->dbHandler->createSelectQuery();
```

Comment:
Well, at first I would probably go with `$this->dbHandler` in the other places, but yeah, I must agree it's  better to use `getConnection` :-)

So, fixup attempt no.2: 9167e50.